src/services/offer.js

```diff
@@ -54,6 +54,65 @@ class OfferService {
             coordinates,
             requirements
         });
+        return offer;
+    }
+
+    async edit(
+        _id,
+        {
+            title,
+            publishDate,
+            publishEndDate,
+            jobMinDuration,
+            jobMaxDuration,
+            jobStartDate,
+            description,
+            contacts,
+            isPaid,
+            vacancies,
+            jobType,
+            fields,
+            technologies,
+            isHidden,
+            location,
+            coordinates,
+            requirements,
+        }) {
+
+        const query = { _id };
+
+        // Cleaning up null entries
+        const edits = Object.fromEntries(
+            Object.entries({
+                title,
+                publishDate,
+                publishEndDate,
+                jobMinDuration,
+                jobMaxDuration,
+                jobStartDate,
+                description,
+                contacts,
+                isPaid,
+                vacancies,
+                jobType,
+                fields,
+                technologies,
+                isHidden,
+                location,
+                coordinates,
+                requirements,
+            }).filter(([_, v]) => v));
+
+        const offer = await Offer.findOneAndUpdate(
+            query,
+            edits,
+            { new: true },
+            (err) => {
+                if (err) {
+                    throw err;
+                }
+            }
+        );
```

Comment:
```suggestion

        const fields = {
                title,
                publishDate,
                publishEndDate,
                jobMinDuration,
                jobMaxDuration,
                jobStartDate,
                description,
                contacts,
                isPaid,
                vacancies,
                jobType,
                fields,
                technologies,
                isHidden,
                location,
                coordinates,
                requirements,
            }
 
       try {
        const offer = await Offer.findByIdAndUpdate(
            _id,
            fields,
            { new: true, omitUndefined:true },
        );
        } catch (err) {
           console.error(err)
           throw err
       }
```

If there is an error here it will be caught in the route file. Just adding logging so we can know if it fails

Reply:
`web-dev_1   | Title:  undefined`
`web-dev_1   | PublishDate:  null`
`web-dev_1   | PublishEndDate:  null`
This causes PublishDate to be edited to null

Reply:
The problem is in the request validator that is not failing with null

Reply:
Actually, there are two things here:
* If this is call by the route logic, it will never be null, since it's always a string, and it is verified by the validators
* If called by any other place, null should actually be allowed, since it is a valid update in terms of the model (it's like unpublishing an offer)

Reply:
Ok, found the problem. Needed to place `.optional` in date sanitizers, otherwise it would try to convert `undefined`